server/src/test/java/org/opensearch/index/mapper/DenseVectorFieldTypeTests.java

```diff
@@ -0,0 +1,531 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.mapper;
+
+import org.junit.Before;
+import org.mockito.Mockito;
+import org.opensearch.common.CheckedConsumer;
+import org.opensearch.common.Strings;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.compress.CompressedXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.index.IndexService;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.QueryShardException;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.opensearch.index.mapper.FieldTypeTestCase.MOCK_QSC_DISALLOW_EXPENSIVE;
+import static org.opensearch.index.mapper.KnnAlgorithmContext.Method.HNSW;
+import static org.opensearch.index.mapper.KnnAlgorithmContextFactory.HNSW_PARAMETER_BEAM_WIDTH;
+import static org.opensearch.index.mapper.KnnAlgorithmContextFactory.HNSW_PARAMETER_MAX_CONNECTIONS;
+
+public class DenseVectorFieldTypeTests extends OpenSearchSingleNodeTestCase {
```

Comment:
Most of these are `FieldMapperTests` (parsing the mapping) not `FieldTypeTests` so lets refactor them over to `DenseVectorFieldMapperTests`.

Also, `OpenSearchSingleNodeTestCase` is slow and big so we try to avoid it. Let's refactor this test class as follows:


```suggestion
public class DenseVectorFieldTypeTests extends FieldTypeTestCase {
```

For guidance on this test see the [CollationFieldTypeTests](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/analysis-icu/src/test/java/org/opensearch/index/mapper/CollationFieldTypeTests.java) as an example.